app/Metadata/Extractor.php

```diff
@@ -56,28 +57,61 @@ public function bare()
 	public function filesize(string $path): int
 	{
 		return (int) filesize($path);
-		/*$size = $filesize_raw / 1024;
-		if ($size >= 1024) {
-			$metadata['filesize'] = round($size / 1024, 1) . ' MB';
-		} else {
-			$metadata['filesize'] = round($size, 1) . ' KB';
-		}*/
 	}
 
 	/**
-	 * Extracts metadata from an image file.
+	 * Returns the SHA-1 checksum of a file.
+	 *
+	 * @param string $path The relative file path
+	 *
+	 * @return string the checksum
+	 */
+	public function checksum(string $path): string
+	{
+		$checksum = sha1_file($path);
+		if ($checksum === false) {
+			$msg = 'Could not compute checksum for: ' . $path;
+			Logs::error(__METHOD__, __LINE__, $msg);
+			throw new \RuntimeException($msg);
+		}
+
+		return $checksum;
+	}
+
+	/**
+	 * Extracts metadata from a file.
+	 *
+	 * **Warning:**
+	 *
+	 * The parameter `$kind` is enum-like parameter and accepts the values
+	 * `photo`, `video` or `raw` (see
+	 * {@link \App\Actions\Photo\Extensions\Checks::file_kind}).
+	 * In other words `kind` is a coarsening of the mime type of a file, but
+	 * not identical to the mime type.
+	 * See {@link \App\Actions\Photo\Create::add()} which sets `$kind` to the
+	 * result of {@link \App\Actions\Photo\Extensions\Checks::file_kind()}.
+	 * However, there are at least three other occurrences where this method
+	 * is called and the full mime type is passed as the second parameter:
+	 * see {@link \App\Console\Commands\ExifLens::handle()},
+	 * {@link \App\Console\Commands\Takedate::handle()} and
+	 * {@link \App\Console\Commands\VideoData::handle()}.
+	 *
+	 * IMHO, there is an amazing number of places which somehow deal with
+	 * "mime type-ish" sort of values with subtle differences.
+	 *
+	 * TODO: Thoroughly refactor this.
```

Comment:
:heavy_plus_sign:

Reply:
Yes, that is another PR. I fear it will never end. :pleading_face: